WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewModel.swift

```diff
@@ -165,6 +177,25 @@ extension ShippingLabelAddressFormViewModel {
         case postcode
         case state
         case country
+        case missingPhoneNumber
+        case invalidPhoneNumber
+    }
+
+    /// Validates phone number for origin address.
+    /// This take into account whether phone is not empty,
+    /// has length 10 with additional "1" area code for US.
+    ///
+    /// Note: This logic may need to be updated if there is a need for validating other cases.
+    ///
+    private var isPhoneNumberValid: Bool {
+        guard let phone = address?.phone, phone.isNotEmpty else {
+            return false
+        }
+        if phone.hasPrefix("1") {
+            return phone.count == 11
+        } else {
+            return phone.count == 10
```

Comment:
I don't know how important this is, but [the web implementation](https://github.com/Automattic/woocommerce-services/blob/f772b6ed36fad5576582c9b8c5c044a6f0b7d2c2/client/extensions/woocommerce/woocommerce-services/state/shipping-label/selectors.js#L246) seems to discard anything that's not a digit from the count, so you can input the phone number with spaces, dashes, or parentheses. This might seem unnecessary because the keyboard type will only allow numeric inputs, but phone numbers formatted differently can still be pasted:

**Web:**

![Screen Shot 2021-08-03 at 08 09 15](https://user-images.githubusercontent.com/8739/127966388-e87845bb-8159-4a75-b058-cb8ebe16855a.png)

**App:**
![Screen Shot 2021-08-03 at 08 13 11](https://user-images.githubusercontent.com/8739/127966818-62949553-ad9a-43d3-81ff-937f72367720.png)

Reply:
Great catch! I think it's safer to filter out special characters so I've done that in 81cbf65488437efdd2f5f38a352ea1a01786cecf.